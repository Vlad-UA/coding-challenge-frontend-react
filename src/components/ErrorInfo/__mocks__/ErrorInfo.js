// Core
import React from 'react';

const ErrorInfoMock = () => <div className="error-info-mock">ErrorInfoMock</div>;
ErrorInfoMock.displayName = 'ErrorInfoMock';
export default ErrorInfoMock;
