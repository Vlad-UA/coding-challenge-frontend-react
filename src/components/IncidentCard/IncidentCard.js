// Core
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Link } from 'react-router-dom';

// Instruments
import styles from './IncidentCard.module.scss';

export default class IncidentCard extends PureComponent {
  render() {
    const {
      incident, incident: {
        title, description, occurred_at, updated_at, media: { image_url_thumb } = {}, address, id,
      },
    } = this.props;

    return (
      <div className={styles.IncidentCard}>
        <div>
          {
            image_url_thumb && <img width={100} height={100} alt="bike" src={image_url_thumb} />
          }
        </div>
        <div className={styles.data}>
          <Link to={{ pathname: `case/${id}`, state: { incident } }}><div>{title}</div></Link>
          <div className={styles.description}>{description}</div>
          <div>
            <strong>Date of the theft:</strong>
            {` ${moment.unix(occurred_at).format('ddd MMM D YYYY')}`}
          </div>
          <div>
            <strong>Report date:</strong>
            {`: ${moment.unix(updated_at).format('ddd MMM D YYYY')}`}
          </div>
          <div>
            <strong>Location:</strong>
            {` ${address}`}
          </div>
        </div>
      </div>
    );
  }
}

IncidentCard.propTypes = {
  incident: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    address: PropTypes.string,
    description: PropTypes.string,
    occurred_at: PropTypes.number,
    updated_at: PropTypes.number,
    media: PropTypes.shape({
      image_url_thumb: PropTypes.string,
    }),
  }).isRequired,
};
