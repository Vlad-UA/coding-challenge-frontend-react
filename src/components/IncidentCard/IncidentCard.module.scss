.IncidentCard{
  display: grid;
  grid-template-columns: 100px auto;
  grid-column-gap: 10px;
  border: solid 2px darkgray;
  padding: 10px;

  .description{
    margin-top: 5px;
  }

  .dateLocation{
    margin-top: 10px;
  }
}
