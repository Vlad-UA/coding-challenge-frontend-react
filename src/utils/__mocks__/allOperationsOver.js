export const allOperationsOver = () => new Promise(resolve => setTimeout(resolve, 0));
