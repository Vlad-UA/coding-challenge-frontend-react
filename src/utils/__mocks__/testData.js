const OCCURRED_AT = 1552325609;

class IncidentData {
  constructor(
    id = Math.floor(10000 + Math.random() * 90000),
    title = `Title ${id}`,
    description = `Description ${id}`,
    address = `Address ${id}`,
    occurred_at = OCCURRED_AT,
    updated_at = 1553542734,
    image_url_thumb = 'https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg',
    type = 'Theft',
  ) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.address = address;
    this.occurred_at = occurred_at;
    this.updated_at = updated_at;
    this.media = {
      image_url_thumb,
    };
    this.type = type;
  }
}

// One Incident
export class IncidentModel {
  constructor({
    id,
    title,
    description,
    address,
    occurred_at,
    updated_at,
    image_url_thumb,
    type,
  }) {
    this.incident = new IncidentData(
      id,
      title,
      description,
      address,
      occurred_at,
      updated_at,
      image_url_thumb,
      type,
    );
  }
}

// Many Incidents
export class IncidentsModel {
  constructor(
    ids = [97572, 94547],
  ) {
    this.incidents = ids.map(id => new IncidentData(id));
  }
}


class LocationData {
  constructor(
    id = Math.floor(10000 + Math.random() * 90000),
    occurred_at = OCCURRED_AT,
  ) {
    this.properties = {
      id,
      occurred_at,
    };
    this.geometry = {
      type: 'Point',
      coordinates: [
        -116.1960257,
        43.6334557,
      ],
    };
  }
}

export class LocationsModel {
  constructor(
    ids = [97572],
  ) {
    this.type = 'FeatureCollection';
    this.features = ids.map(id => new LocationData(id));
  }
}
