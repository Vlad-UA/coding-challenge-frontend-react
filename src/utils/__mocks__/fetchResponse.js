// Test utilities
import { IncidentsModel } from './testData';

export class FetchResponse {
  constructor({
    status = global.fetchResponseStatus,
    fetchResponseData = new IncidentsModel(),
  } = {}) {
    return {
      status,
      json() {
        return Promise.resolve(fetchResponseData);
      },
    };
  }
}
