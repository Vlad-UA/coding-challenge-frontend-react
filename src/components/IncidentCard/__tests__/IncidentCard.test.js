/* eslint-env jest */

// Core
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

// Test utilities
import renderer from 'react-test-renderer';
import { IncidentModel } from '../../../utils/__mocks__/testData';

// Component
import IncidentCard from '../IncidentCard';

describe('Component IncidentCard:', () => {
  test('Must match snapshot:', () => {
    const { incident } = new IncidentModel({ id: 83629 });

    const incidentCard = renderer
      .create(
        <BrowserRouter>
          <IncidentCard incident={incident} />
        </BrowserRouter>,
      )
      .toJSON();

    expect(incidentCard).toMatchSnapshot();
  });
});
