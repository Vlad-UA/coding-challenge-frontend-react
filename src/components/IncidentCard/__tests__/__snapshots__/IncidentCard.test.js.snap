// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component IncidentCard: Must match snapshot: 1`] = `
<div
  className="IncidentCard"
>
  <div>
    <img
      alt="bike"
      height={100}
      src="https://files.bikeindex.org/uploads/Pu/148912/small_IMG_20170705_191324.jpg"
      width={100}
    />
  </div>
  <div
    className="data"
  >
    <a
      href="case/83629"
      onClick={[Function]}
    >
      <div>
        Title 83629
      </div>
    </a>
    <div
      className="description"
    >
      Description 83629
    </div>
    <div>
      <strong>
        Date of the theft:
      </strong>
       Mon Mar 11 2019
    </div>
    <div>
      <strong>
        Report date:
      </strong>
      : Mon Mar 25 2019
    </div>
    <div>
      <strong>
        Location:
      </strong>
       Address 83629
    </div>
  </div>
</div>
`;
